### PR TITLE
Add `additionalEnv` helm settings

### PR DESCRIPTION
**Pull request by alpeb**
Add `additionalEnv` helm settings to the proxy and controller manifests alongside the existing `experimentalEnv` ones.

### CODE REVIEW

File: charts/linkerd-control-plane/templates/destination.yaml [hunk 1]
```diff
@@ -208,10 +208,15 @@ spec:
         {{- range (.Values.destinationController).experimentalArgs }}
         - {{ . }}
         {{- end }}
-        {{- with (.Values.destinationController).experimentalEnv }}
```
Comment (olix0r) [inline charts/linkerd-control-plane/templates/destination.yaml:211]: tioli, I think we could minimize this with something like
```
        {{- with .Values.destinationController | concat .experimentalEnv .additionalEnv }}
```

I haven't tested this specific syntax but the general idea should avoid duplication.  Here and throughout.  Furthermore, we could consider using a partial here to reduce boilerplate

**Review by olix0r — APPROVED** [review]
non-blocking suggestions

File: charts/linkerd-control-plane/templates/destination.yaml [hunk 2]
```diff
@@ -208,10 +208,15 @@ spec:
...
+        {{- if or (.Values.destinationController).additionalEnv (.Values.destinationController).experimentalEnv }}
         env:
+        {{- with (.Values.destinationController).additionalEnv }}
         {{- toYaml . | nindent 8 -}}
         {{- end }}
+        {{- with (.Values.destinationController).experimentalEnv }}
+        {{- toYaml . | nindent 8 -}}
+        {{- end }}
+        {{- end }}
```
Comment (olix0r) [inline charts/linkerd-control-plane/templates/destination.yaml:219]: tioli, I think we could minimize this with something like
```
        {{- with .Values.destinationController | concat .experimentalEnv .additionalEnv }}
```

I haven't tested this specific syntax but the general idea should avoid duplication.  Here and throughout.  Furthermore, we could consider using a partial here to reduce boilerplate

**Review by adleong — APPROVED** [review]

File: charts/linkerd-control-plane/templates/destination.yaml (hunk 2 above)
Comment (alpeb) [reply charts/linkerd-control-plane/templates/destination.yaml:219]: It appears `with` cannot be used in a pipe. So we have to replicate `.Values.destinationController` on the right. And concat will error if one of its args is nil, so we have to pipe to `default list ""` both vars. So the one liner ends up being not pretty.

**Pull request merged**